lib/src/message.dart

```diff
@@ -12,11 +12,12 @@ import 'body.dart';
 import 'headers.dart';
 import 'http_unmodifiable_map.dart';
 
+/// Retrieves the [Body] contained in the [message].
 Body getBody(Message message) => message._body;
 
 /// The default set of headers for a message created with no body and no
 /// explicit headers.
-final _defaultHeaders = new HttpUnmodifiableMap<String>({"content-length": "0"},
+final _defaultHeaders = new HttpUnmodifiableMap<String>({},
```

Comment:
I don't think just making this empty is enough. We still need to send `content-length: 0` for `null`-bodied POSTs and other requests that have content.